lib/methods/ifwapichecktask.cpp

```diff
@@ -498,20 +429,28 @@ void IfwApiCheckTask::ScriptFunc(const Checkable::Ptr& checkable, const CheckRes
 	auto& io (IoEngine::Get().GetIoContext());
 	auto strand (Shared<asio::io_context::strand>::Make(io));
 	Shared<asio::ssl::context>::Ptr ctx;
-	double start = Utility::GetTime();
+
+	cr->SetExecutionStart(Utility::GetTime());
+	cr->SetCommand(cmdLine);
 
 	try {
 		ctx = SetupSslContext(cert, key, ca, crl, DEFAULT_TLS_CIPHERS, DEFAULT_TLS_PROTOCOLMIN, DebugInfo());
 	} catch (const std::exception& ex) {
-		ReportIfwCheckResult(checkable, cmdLine, cr, ex.what(), start, Utility::GetTime());
+		cr->SetOutput(ex.what());
+		cr->SetExecutionEnd(Utility::GetTime());
+
+		resultHandler(cmdLine, ProcessResult{});
 		return;
 	}
 
 	auto conn (Shared<AsioTlsStream>::Make(io, *ctx, expectedSan));
 
+	// Handler for the final check result that is used to asynchronously process within the global non-I/O thread pool.
+	auto handler ([cmdLine, resultHandler = std::move(resultHandler)]() { resultHandler(cmdLine, ProcessResult{}); });
```

Comment:
I'm not sure what the benefit is of defining that lambda here instead of simply where `Utility::QueueAsyncCallback()` is called.

Reply:
To avoid unnecessary copy of the `resultHandler` lambda, as I wouldn't be able to move it into the `Utility::QueueAsyncCallback()` callback from within the coroutine.